### PR TITLE
Use DefaultRazorDiagnostic in DefaultDirectiveSyntaxTreePass

### PR DESCRIPTION
**Pull request by ajaybhargavb**
#1827 

This is just one of many usages that must be moved. Just sent this one first to make sure every one is on the same page as to what needs to be done. More PRs will follow this one.

FYI @mkArtakMSFT

### CODE REVIEW

File: src/Microsoft.AspNetCore.Razor.Language/RazorDiagnosticFactory.cs [hunk 1]
```diff
@@ -55,6 +55,16 @@ public static RazorDiagnostic CreateParsing_DuplicateDirective(string directive,
             return RazorDiagnostic.Create(Parsing_DuplicateDirective, location, directive);
         }
 
+        internal static readonly RazorDiagnosticDescriptor Parsing_SectionsCannotBeNested =
+            new RazorDiagnosticDescriptor(
+                $"{DiagnosticPrefix}2002",
+                () => LegacyResources.ParseError_Sections_Cannot_Be_Nested,
```
Comment (ajaybhargavb) [inline src/Microsoft.AspNetCore.Razor.Language/RazorDiagnosticFactory.cs]: Should this move from `LegacyResources` to `Resources`?

Comment (NTaylorMullen) [reply src/Microsoft.AspNetCore.Razor.Language/RazorDiagnosticFactory.cs]: Yes

Comment (ajaybhargavb) [reply src/Microsoft.AspNetCore.Razor.Language/RazorDiagnosticFactory.cs]: I'll keep then in LegacyResources for now. I'll move them all together in a separate PR once we get rid of the legacy diagnostic.

Comment (rynowak) [reply src/Microsoft.AspNetCore.Razor.Language/RazorDiagnosticFactory.cs]: yes, these should eventually move. The goal is to nuke `LegacyResources`

**Pull request merged**